src/vm/proftoeeinterfaceimpl.cpp

```diff
@@ -6750,6 +6754,63 @@ HRESULT ProfToEEInterfaceImpl::GetCodeInfo4(UINT_PTR pNativeCodeStartAddress,
                                     codeInfos);
 }
 
+/*
+ * GetObjectReferences
+ * 
+ * Gets the object references (if any) from the ObjectID.
+ * 
+ * Parameters:
+ *      objectId        - object id of interest
+ *      cNumReferences  - count of references for which the profiler has allocated buffer space
+ *      pcNumReferences - actual count of references
+ *      references      - filled array of object references
+ *
+ * Returns:
+ *   S_OK if successful
+ *
+ */
+HRESULT ProfToEEInterfaceImpl::GetObjectReferences(ObjectID objectId, ULONG32 cNumReferences, ULONG32 *pcNumReferences, ObjectID references[])
+{
+    CONTRACTL
+    {
+        NOTHROW;
+        GC_NOTRIGGER;
+        MODE_ANY;
+        EE_THREAD_NOT_REQUIRED;
+        CANNOT_TAKE_LOCK;
+    }
+    CONTRACTL_END;
+
+    PROFILER_TO_CLR_ENTRYPOINT_SYNC_EX(
+        kP2EEAllowableAfterAttach,
+        (LF_CORPROF,
+        LL_INFO1000,
+        "**PROF: GetObjectReferences 0x%p.\n",
+        objectId));
+
```

Comment:
Could you add a check that returnes E_INVALIDARG if  cNumReferences > 0 and pcNumReferences or references is null? I realize that we haven't been extremely consistent with defensive programming in the profiler interfaces but it's something I'd like to make us be better about.